itest/performance/perf_test.go

```diff
@@ -0,0 +1,49 @@
+//go:build perf_test
+
+package perf_test
+
+import (
+	"context"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestPerformance executes the configured performance tests.
+func TestPerformance(t *testing.T) {
+	cfg, err := LoadConfig()
+	require.NoError(t, err, "unable to load main config")
+
+	//
+	suiteTimeout := time.Duration(cfg.TestSuiteTimeout) * time.Minute
+	ctxt, cancel := context.WithTimeout(
+		context.Background(), suiteTimeout,
+	)
+	defer cancel()
+
+	// Execute each one test case.
+	// TODO(positiveblue): have a way to run them in parallel?
+	testList := strings.Split(cfg.TestCases, ",")
+	for _, testCase := range testList {
+		execTestCase(t, ctxt, testCase, cfg)
+	}
+}
+
+// execTestCase is the method in charge of executing a single test case.
+func execTestCase(t *testing.T, ctx context.Context, testName string,
+	cfg *Config) {
+
+	testTimeout := time.Duration(cfg.TestTimeout) * time.Minute
+	ctxt, cancel := context.WithTimeout(ctx, testTimeout)
+	defer cancel()
+
+	switch testName {
+	case "mint_batch_stress":
```

Comment:
We could use the same mechanism as we already do in the itest where we define all test cases as a list and run them as sub tests if it is enabled by the config. But just an idea, non-blocking.

Reply:
this is a good point, I will update it to follow the same pattern before merging